HW1/P5/P5.txt

```diff
@@ -1 +1,130 @@
+[u'Harvard_University', u'The_Da_Vinci_Code', u'Thriller_(genre)', u'Kevin_Bacon']
+[u'Harvard_University', u'Jack_Lemmon', u'Mister_Roberts_(film)', u'Kevin_Bacon']
+[u'Harvard_University', u'Tommy_Lee_Jones', u'2005_Cannes_Film_Festival', u'Kevin_Bacon']
+[u'Harvard_University', u'Hasty_Pudding_Man_of_the_Year', u'Tom_Hanks', u'Kevin_Bacon']
+[u'Harvard_University', u'Dan_Brown', u'Tom_Hanks', u'Kevin_Bacon']
+[u'Harvard_University', u'Hasty_Pudding_Theatricals', u'Tom_Hanks', u'Kevin_Bacon']
+[u'Harvard_University', u"Conan_O'Brien", u'Tom_Hanks', u'Kevin_Bacon']
+[u'Harvard_University', u'The_Da_Vinci_Code', u'Tom_Hanks', u'Kevin_Bacon']
+[u'Harvard_University', u'Natalie_Portman', u'New_York,_I_Love_You', u'Kevin_Bacon']
+[u'Harvard_University', u'John_Kerry', u'John_Edwards_presidential_campaign,_2008', u'Kevin_Bacon']
+[u'Harvard_University', u'September_16', u'Mickey_Rourke', u'Kevin_Bacon']
+[u'Harvard_University', u'Boston_Brahmin', u'Endicott_Peabody_(educator)', u'Kevin_Bacon']
+[u'Harvard_University', u'Franklin_D._Roosevelt', u'Endicott_Peabody_(educator)', u'Kevin_Bacon']
+[u'Harvard_University', u'Boston,_Massachusetts', u'South_Boston,_Boston,_Massachusetts', u'Kevin_Bacon']
+[u'Harvard_University', u'Longwood_Medical_and_Academic_Area', u'South_Boston,_Boston,_Massachusetts', u'Kevin_Bacon']
+[u'Harvard_University', u'Charlestown,_Massachusetts', u'South_Boston,_Boston,_Massachusetts', u'Kevin_Bacon']
+[u'Harvard_University', u'Jamaica_Plain,_Massachusetts', u'South_Boston,_Boston,_Massachusetts', u'Kevin_Bacon']
+[u'Harvard_University', u'2006', u'2006_in_film', u'Kevin_Bacon']
+[u'Harvard_University', u'University_of_Hartford', u'Apollo_13', u'Kevin_Bacon']
+[u'Harvard_University', u'Richard_Nixon', u'Apollo_13', u'Kevin_Bacon']
+[u'Harvard_University', u'London', u'Elvis_Costello', u'Kevin_Bacon']
+[u'Harvard_University', u'Audioslave', u'Elvis_Costello', u'Kevin_Bacon']
+[u'Harvard_University', u'University_of_Virginia', u'Tiki_Barber', u'Kevin_Bacon']
+[u'Harvard_University', u'John_Lithgow', u'Footloose', u'Kevin_Bacon']
+[u'Harvard_University', u'Gilmore_Girls', u'Liza_Weil', u'Kevin_Bacon']
+[u'Harvard_University', u'The_New_York_Times', u'Judith_Miller_(journalist)', u'Kevin_Bacon']
+[u'Harvard_University', u'Matt_Damon', u'Massachusetts_State_Police', u'Kevin_Bacon']
+[u'Harvard_University', u'Plymouth,_Massachusetts', u'Massachusetts_State_Police', u'Kevin_Bacon']
+[u'Harvard_University', u'Texas_A&M_University', u'A_Few_Good_Men', u'Kevin_Bacon']
+[u'Harvard_University', u'Harvard_Law_School', u'A_Few_Good_Men', u'Kevin_Bacon']
+[u'Harvard_University', u'Al_Gore', u'Live_Earth_(2007_concert)', u'Kevin_Bacon']
+[u'Harvard_University', u'Cornell_University', u'Edmund_Bacon', u'Kevin_Bacon']
+[u'Harvard_University', u'2003', u'2003_in_film', u'Kevin_Bacon']
+[u'Harvard_University', u'Legally_Blonde', u'2003_in_film', u'Kevin_Bacon']
+[u'Harvard_University', u'Thomas_Dudley', u'Kyra_Sedgwick', u'Kevin_Bacon']
```

Comment:
I guess you did not attempt connected components; oh well, I think you could have gotten it! Your code is great.